### PR TITLE
enhancement(routine-table): added support for crn in routing table

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM1**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_subnet_routing_table_attachment.go [hunk 1]
```diff
@@ -233,6 +255,30 @@ func resourceIBMISSubnetRoutingTableAttachmentUpdate(context context.Context, d
 		return resourceIBMISSubnetRoutingTableAttachmentRead(context, d, meta)
 	}
 
+	if d.HasChange(isRoutingTableCrn) {
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_subnet_routing_table_attachment.go:258]: verify if this handles the case where user removes crn and adds id, or vice versa

File: ibm/service/vpc/resource_ibm_is_subnet.go [hunk 2]
```diff
@@ -613,6 +635,14 @@ func subnetUpdate(d *schema.ResourceData, meta interface{}, id string) error {
 			return err
 		}*/
 	}
+	if d.HasChange(isSubnetRoutingTableCrn) {
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_subnet.go:638]: please verify if this handles the case where crn is removed and if is added and vice versa

File: ibm/service/vpc/data_source_ibm_is_vpc_routing_tables.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ import (
 const (
 	isRoutingTableAcceptRoutesFrom      = "accept_routes_from"
 	isRoutingTableID                    = "routing_table"
+	isRoutingTableCrn                   = "routing_table_crn"
```
Comment (deepaksibm) [inline ibm/service/vpc/data_source_ibm_is_vpc_routing_tables.go:20]: set the value in this data source.

**Comment by SunithaGudisagarIBM1** [comment]
```
	data "ibm_is_vpc_routing_tables" "test1" {
		vpc =  ibm_is_vpc.testacc_vpc.id
	}
```
Screenshot 2024-08-04 at 5 21 50 PM


```
resource "ibm_is_vpc_routing_table" "test_route_rt" {
  vpc                           = ibm_is_vpc.testacc_vpc.id
  name                          = "test-vpc-routing-table"
  route_direct_link_ingress     = true
  route_transit_gateway_ingress = false
  route_vpc_zone_ingress        = false
}
```
Screenshot 2024-08-04 at 5 22 54 PM

```
data "ibm_is_vpc_routing_table" "ibm_is_vpc_routing_table_name" {
  vpc  = ibm_is_vpc.testacc_vpc.id
  name = ibm_is_vpc_routing_table.test_route_rt.name
}
```
Screenshot 2024-08-04 at 5 23 18 PM

```
data "ibm_is_vpc_default_routing_table" "def_route_table" {
  vpc = ibm_is_vpc.testacc_vpc.id
}
```
Screenshot 2024-08-04 at 5 24 27 PM

```
data "ibm_is_subnets" "ds_subnets_resource_group" {
}
```
Screenshot 2024-08-04 at 5 25 05 PM

```
data "ibm_is_subnet" "ds_subnet" {
  identifier = "0716-f3c6cdb5-7923-461f-826b-f1b4f1189970"
}
```
Screenshot 2024-08-04 at 5 25 41 PM

File: ibm/service/vpc/resource_ibm_is_subnet.go (hunk 2 above)
Comment (SunithaGudisagarIBM1) [reply ibm/service/vpc/resource_ibm_is_subnet.go:638]: resource "ibm_is_subnet" "testacc_subnet" {
  name            = "terrraform-test"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
  # routing_table = ibm_is_vpc.testacc_vpc.default_routing_table
  # routing_table_crn = ibm_is_vpc.testacc_vpc.default_routing_table_crn
  routing_table_crn = ibm_is_vpc_routing_table.test_route_rt.crn
  # routing_table = ibm_is_vpc_routing_table.test_route_rt.routing_table
}

same routing table/crn no diff, different id or crn will result into force replace.

Screenshot 2024-08-04 at 6 42 57 PM

**Comment by SunithaGudisagarIBM1** [comment]
```
resource "ibm_is_subnet" "testacc_subnet" {
  name            = "terrraform-test"
  vpc             = ibm_is_vpc.testacc_vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
 -> routing_table = ibm_is_vpc.testacc_vpc.default_routing_table
 -> routing_table_crn = ibm_is_vpc.testacc_vpc.default_routing_table_crn
  routing_table_crn = ibm_is_vpc_routing_table.test_route_rt.crn
  -> routing_table = ibm_is_vpc_routing_table.test_route_rt.routing_table
}
```
Screenshot 2024-08-04 at 6 44 38 PM

**Comment by SunithaGudisagarIBM1** [comment]
```
resource "ibm_is_vpc_routing_table" "test_route_rt" {
  vpc                           = ibm_is_vpc.testacc_vpc.id
  name                          = "test-vpc-routing-table"
  route_direct_link_ingress     = false
  route_transit_gateway_ingress = false
  route_vpc_zone_ingress        = false
}
```
Screenshot 2024-08-04 at 6 45 17 PM

File: ibm/service/vpc/resource_ibm_is_subnet_routing_table_attachment.go (hunk 1 above)
Comment (SunithaGudisagarIBM1) [reply ibm/service/vpc/resource_ibm_is_subnet_routing_table_attachment.go:258]:
```
resource "ibm_is_subnet_routing_table_attachment" "attach" {
  # depends_on    = [ibm_is_vpc_routing_table.testacc_vpc_routing_table, ibm_is_subnet.testacc_subnet]
  subnet = "0726-9d4bd36c-5d9f-4494-beed-15d5ed0f021d"
  routing_table   = ibm_is_vpc.testacc_vpc.default_routing_table
  # routing_table = ibm_is_vpc_routing_table.test_route_rt.routing_table
  # routing_table_crn = ibm_is_vpc_routing_table.test_route_rt.crn
}
```
Screenshot 2024-08-04 at 8 32 16 PM

Comment (SunithaGudisagarIBM1) [reply ibm/service/vpc/resource_ibm_is_subnet_routing_table_attachment.go:258]: same routing table/crn no diff, different id or crn will result into force replace.

**Comment by astha-jain** [comment]
Missing resource_group - ?

**Comment by SunithaGudisagarIBM1** [comment]
Prod Testing Results
```
resource "ibm_is_vpc" "testacc_vpc" {
  name = "terraform-test-1"
}

data "ibm_is_vpc_default_routing_table" "def_route_table" {
  vpc = ibm_is_vpc.testacc_vpc.id
}
```
Screenshot 2024-08-28 at 11 23 19 PM
Screenshot 2024-08-28 at 11 23 38 PM

**Comment by SunithaGudisagarIBM1** [comment]
```
	data "ibm_is_vpc_routing_tables" "test1" {
		vpc =  ibm_is_vpc.testacc_vpc.id
	}
```
Screenshot 2024-08-28 at 11 25 47 PM

**Comment by SunithaGudisagarIBM1** [comment]
```
resource "ibm_is_vpc_routing_table" "test_route_rt" {
  vpc                           = ibm_is_vpc.testacc_vpc.id
  name                          = "test-vpc-routing-table"
  route_direct_link_ingress     = true
  route_transit_gateway_ingress = false
  route_vpc_zone_ingress        = false
}
```
Screenshot 2024-08-28 at 11 27 48 PM

**Comment by SunithaGudisagarIBM1** [comment]
```
data "ibm_is_vpc_routing_table" "ibm_is_vpc_routing_table_name" {
  vpc  = ibm_is_vpc.testacc_vpc.id
  name = ibm_is_vpc_routing_table.test_route_rt.name
}
```
Screenshot 2024-08-28 at 11 29 25 PM

**Comment by SunithaGudisagarIBM1** [comment]
@astha-jain updated with resource group attribute where ever it is applicable, please review

**Review by astha-jain — APPROVED** [review]
Please move to upstream

**Pull request closed without merging**